src/TokensTracker/Trackers/NEP-11/Nep11Tracker.cs

```diff
@@ -92,7 +92,18 @@ public override void OnPersist(NeoSystem system, Block block, DataCache snapshot
                         Log($"{state.Hash} is not nft!", LogLevel.Warning);
                         continue;
                     }
-
+                    if (balanceMethod != null &&
+                        !(balanceMethod.Parameters[0].Type == ContractParameterType.Hash160 && balanceMethod.ReturnType == ContractParameterType.Integer))
+                    {
+                        Log($"{state.Hash} is not nft:balance1!", LogLevel.Warning);
+                        continue;
+                    }
+                    if (balanceMethod2 != null &&
```

Comment:
should be an `else if` Because the function is one or the other. Also if the token has both functions you don't check for that either. As the code sits right now, if 1st function exists look for other function. Which isn't valid to what NEP-11 says
https://github.com/neo-project/proposals/blob/master/nep-11.mediawiki#user-content-decimals

Reply:
I think also balance1 and balance2 is a bad naming.
Let's say something like "balanceInteger" and "balanceDivisible"

Reply:
Also there is no need to check to `abi` for these functions.

@vncoelho I think `balanceDivisible` and `balanceNonDivisible` would fit better.